File: NodeJS/rs-example.js
/**
 * NAME: JS script example
 * SYNOPSIS: Example JS script for RunScript GUI with commander library usage
 * TAGS: Example "RunScript Team"
 */

// Import 'program' and 'Option' from 'commander'. This library enables
// defining command-line options and parameters. For more details, see
// 'commander' documentation: https://github.com/tj/commander.js/
const { program, Option } = require("commander");

program
  .option("-dp, --danger-param <type>", "Set the danger level parameter{danger}", "default")
  .option("-wp, --warning-param <type>", "Set the warning level parameter{warning}")
  .option("-lp, --long-param <description>", "Set a parameter with an extended description")
  .option("-js, --javascript-script <identifier>", "Specify a JavaScript snippet with its ID{id:rs-js-subscript-example}")
  .option("-ps, --powershell-script <identifier>", "Specify a PowerShell script with its ID{id:rs-ps-subscript-example}")
  .option("-py, --python-script [identifier]", "Specify a Python script with its ID, optional{id:rs-python-subscript-example}")
  .option("-bs, --bash-script [identifier]", "Specify a Bash script with its ID, optional{id:rs-bash-subscript-example}")
  .option("-bx, --boolean-flag", "Set a boolean flag for additional processing")
  .addOption(new Option('-tm, --timeout <seconds>', 'Set the timeout duration in seconds').default(60, '60 seconds'))
  .addOption(new Option('-ds, --drink-size <size>', 'Specify the drink size').choices(['small', 'medium', 'large']))
  .addOption(new Option('-dm, --drink-multiple <sizes...>', 'Specify multiple drink sizes').choices(['small', 'medium', 'large']));
  
program.parse(process.argv);

const options = program.opts();

console.log(options);

File: Python/rs-example.py
"""
NAME: Python script example
SYNOPSIS: Example Python script for RunScript GUI with commander library usage
WARNING: This is a warning message.
DANGER: This is a danger message.
TAGS: Example "RunScript Team" 
"""

import argparse


# When defining argparse, 'prog', 'description', and 'epilog' parameters in
# 'ArgumentParser' provide program info. If 'NAME', 'SYNOPSIS', and
# 'DESCRIPTION' metadata are at the script's start, they should take precedence.
parser = argparse.ArgumentParser(
    prog="Test Script",
    description="Demonstration of using various argument types with argparse.",
    epilog="This is an epilogue containing additional information about the program."
)

# Adding arguments
parser.add_argument("firstname", type=str, default="Martin1", help="Enter the first name")
parser.add_argument("lastname", type=str, default="Spanek", help="Enter the last name {id:test-substring-js}")
parser.add_argument("--string", type=str, required=True, default="Martin", help="String example")
parser.add_argument("--integer", type=int,default=42 , help="Integer example")
parser.add_argument("--float", type=float, help="Floating point number example.")
parser.add_argument("--file", type=argparse.FileType('r'), help="Path to the file for reading")
parser.add_argument("--boolean", type=bool, default=True, help="Boolean value example")
parser.add_argument('--verbose', action='store_true', help='If this switch is used, sets verbose to True.')

# Process the arguments
args = parser.parse_args()

# Print the arguments
print(f"Arguments: {args}")
